clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -253,7 +253,11 @@ public Node nodeById(int id) {
     public Optional<Node> nodeIfOnline(TopicPartition partition, int id) {
         Node node = nodeById(id);
         PartitionInfo partitionInfo = partition(partition);
-        if (node != null && partitionInfo != null && !Arrays.asList(partitionInfo.offlineReplicas()).contains(node)) {
+
+        if (node != null && partitionInfo != null &&
+            !Arrays.asList(partitionInfo.offlineReplicas()).contains(node) &&
```

Comment:
We probably don't need to fix it here, but maybe we should create a little helper to iterate the array instead of creating a new list.